### PR TITLE
Add deprecation flag for context.done()

### PR DESCRIPTION
**Pull request by hossam-nasr**
Adds a `@deprecated` flag for `context.done()`, and adds comments to remove deprecation linting errors.

Question: is this considered a breaking change? If a customer was also using the deprecation linting flag, this would break previously working code.

### CODE REVIEW

File: src/eventHandlers/invocationRequest.ts
Comment (ejizba) [inline src/eventHandlers/invocationRequest.ts]: Any chance we can get https://github.com/Azure/azure-functions-nodejs-worker/pull/547 merged first? This code is completely changing

File: types/index.d.ts
Comment (ejizba) [inline types/index.d.ts:80]: > Question: is this considered a breaking change? If a customer was also using the deprecation linting flag, this would break previously working code.

No, the point of deprecating something is to give users a heads up _before_ you make a breaking change. We currently have no intention of removing `context.done`, but marking it as deprecated now gives users _plenty_ of time to change their code in case we ever decide to remove it.

Also, lint rules are purely optional and we don't consider them for breaking changes.

File: src/eventHandlers/invocationRequest.ts
Comment (hossam-nasr) [reply src/eventHandlers/invocationRequest.ts]: Sure, I'm happy to wait on merging this PR until #547 gets merged. I tried to review #547 but honestly I don't super follow what's going on, so I'd rather defer until @alrod can take a look.

File: types/index.d.ts
Comment (hossam-nasr) [reply types/index.d.ts:80]: Great, thank you!

**Review by ejizba — APPROVED** [review]

**Pull request merged**
